packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -191,14 +197,42 @@ export class DataStores implements IDisposable {
                 }),
             pkg);
 
-        // Resolve pending gets and store off any new ones
         this.contexts.addBoundOrRemoted(remotedFluidDataStoreContext);
 
         // Equivalent of nextTick() - Prefetch once all current ops have completed
         // eslint-disable-next-line @typescript-eslint/no-floating-promises
         Promise.resolve().then(async () => remotedFluidDataStoreContext.realize());
     }
 
+    public processAliasMessage(message: ISequencedDocumentMessage): IDataStoreAliasMapping | undefined {
+        const aliasMessage = message.contents as IDataStoreAliasMessage;
+
+        this.buildAliasMap();
+        const existingMapping = this.aliasMap.get(aliasMessage.alias);
+        if (existingMapping !== undefined) {
+            return {
+                suppliedInternalId: aliasMessage.internalId,
+                alias: aliasMessage.alias,
+                aliasedInternalId: existingMapping,
+            };
+        }
+
+        const currentId = this.aliasMap.get(aliasMessage.internalId);
```

Comment:
why are we getting the internal id from the map here, but the alias above? it seems like the map should be from alias to internal id

Reply:
maybe some comments would help here

Reply:
i know in the alias map we map all datastores to a pseudo alias for their own internal id, but that should never happen on a message

Reply:
I'll make this clearer.